lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -141,6 +146,10 @@ def command=(value)
     task.application_name = value
   end
 
+  def comment=(value)
+    task.comment = value
```

Comment:
I think adding length checks to the various properties is worthy of a separate PR (and should be tagged with PUP-3768). Also I do think you want to `fail` if the length is exceeded. If you only `warn` and the value is truncated, then puppet will keep thinking the resource is out-of-sync.

The limits your proposed sound good. Perhaps for command, we follow what CreateProcess does for its lpCommandLine parameter, "The maximum length of this string is 32,768 characters, including the Unicode terminating null character."